osquery/core/database/tests/rocksdb_database_tests.cpp

```diff
@@ -0,0 +1,118 @@
+/**
+ *  Copyright (c) 2018-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <gtest/gtest.h>
+
+#include <boost/filesystem.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include <osquery/core/database/rocksdb_database.h>
+#include <osquery/database.h>
+
+namespace osquery {
+
+class RocksdbDatabaseTest : public ::testing::Test {
+ protected:
+  std::string path_;
+
+  virtual void SetUp() {
+    boost::uuids::uuid uuid = boost::uuids::random_generator()();
+    auto random_name = boost::uuids::to_string(uuid);
+    auto path = boost::filesystem::temp_directory_path().append(random_name);
+    boost::filesystem::create_directory(path);
+    path_ = path.string();
+  }
+
+  virtual void TearDown() {
+    boost::filesystem::remove_all(path_);
+  }
+};
+
+std::string randomDBPath() {
+  boost::uuids::uuid uuid = boost::uuids::random_generator()();
+  auto random_name = boost::uuids::to_string(uuid);
+  auto path = boost::filesystem::temp_directory_path().append(random_name);
+  boost::filesystem::create_directory(path);
+  return path.string();
+}
+
+TEST_F(RocksdbDatabaseTest, test_open) {
+  auto path = randomDBPath();
+  auto db = std::make_unique<RocksdbDatabase>("test", path_);
+  auto result = db->open();
+  EXPECT_TRUE(result);
+  db->close();
+}
+
+TEST_F(RocksdbDatabaseTest, test_destroy) {
+  auto db = std::make_unique<RocksdbDatabase>("test", path_);
+  EXPECT_TRUE(db->open());
+  EXPECT_TRUE(db->putInt32(kPersistentSettings, "key", 10));
+  db->close();
+  auto result = db->destroyDB();
+  EXPECT_TRUE(result);
+  EXPECT_TRUE(db->open());
+  auto get_result = db->getInt32(kPersistentSettings, "key");
+  EXPECT_FALSE(get_result);
+  EXPECT_EQ(get_result.getError(), DatabaseError::KeyNotFound);
+}
+
+TEST_F(RocksdbDatabaseTest, test_put) {
+  auto db = std::make_unique<RocksdbDatabase>("test", path_);
+  auto result = db->putInt32("test", "test", 23);
+  EXPECT_FALSE(result);
+  EXPECT_EQ(result.getError(), DatabaseError::DbIsNotOpen);
+  EXPECT_TRUE(db->open());
```

Comment:
Maybe it is better to use `ASSERT_TRUE` here instead of `EXPECT_TRUE`. Because if open has failed check putting data to it does not make any sense, right?